src/Sulu/Component/Content/Types/Rlp/Strategy/RlpStrategy.php

```diff
@@ -236,35 +264,35 @@ private function adaptResourceLocators(
                 $property = $structure->getPropertyByTagName('sulu.rlp');
                 $contentType = $this->contentTypeManager->get($property->getContentTypeName());
                 $property->setValue($rlp);
+                $childDocument->setResourceSegment($rlp);
```

Comment:
That's necessary for the moment, so that the recursion still works... Actually it should only be this line, without the `setValue` on the property.